gson/src/test/java/com/google/gson/functional/ReflectionAccessFilterTest.java

```diff
@@ -81,12 +80,12 @@ public void testBlockInaccessibleJava() throws ReflectiveOperationException {
 
     // But serialization should succeed for classes with only public fields.
     // Not many JDK classes have mutable public fields, thank goodness, but java.awt.Point does.
-    Class<?> pointClass = null;
+    Class<?> pointClass;
     try {
       pointClass = Class.forName("java.awt.Point");
     } catch (ClassNotFoundException ignored) {
+      return; // If not found then we don't have AWT and the rest of the test can be skipped.
     }
-    assumeNotNull(pointClass);
```

Comment:
What was the Error Prone warning here?

`return;` makes it look as if this test passed, whereas JUnit's `assume...` will mark it as skipped. I guess the latter is rather desired here.

Reply:
The warning is probably from https://errorprone.info/bugpattern/EmptyCatch, which (along with the Google Style Guide, though that update hasn't been published externally yet) recently got stricter: https://github.com/google/error-prone/pull/4576. You'll eventually see the "Exception" removed from https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions. (That's also what finally got me to go through with https://github.com/google/guava/pull/7395 and some similar PRs to come.)

I agree that an assumption makes more conceptual sense. (But I say this knowing that we consistently just `return` in Guava's tests :))

Reply:
JUnit 4 has `assumeNoException(Throwable)`, but using it in the `catch` seems a bit weird since the exception will then always be non-null, so the assumption always fails.

Maybe it would be cleaner to directly throw an `AssumptionViolatedException`?
```java
Class<?> pointClass;
try {
  pointClass = Class.forName("java.awt.Point");
} catch (ClassNotFoundException e) {
  // If not found then we don't have AWT and the rest of the test can be skipped.
  throw new AssumptionViolatedException("AWT class not present", e);
}
```

Reply:
I still feel that "assume" is not right given that this is happening half way through the test method. The right thing to do is to split out the remainder of the test into a separate method. I will look into doing that in a separate PR.